backend/hidrive/hidrive.go

```diff
@@ -0,0 +1,1021 @@
+			return fs.ErrorObjectNotFound
+		}
+		return err
+	}
+	return o.setMetadata(info)
+}
+
+// ModTime returns the modification time of the object.
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	err := o.readMetadata(ctx)
+	if err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return time.Now()
+	}
+	return o.modTime
+}
+
+// SetModTime sets the metadata on the object to set the modification date.
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	parameters := api.NewQueryParameters()
+	resolvedPath := o.fs.resolvePath(o.remote)
+	parameters.SetPath(resolvedPath)
+	err := parameters.SetTime("mtime", modTime)
+
+	if err != nil {
+		return err
+	}
+
+	opts := rest.Opts{
+		Method:     "PATCH",
+		Path:       "/meta",
+		Parameters: parameters.Values,
+		NoResponse: true,
+	}
+
+	var resp *http.Response
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return o.fs.shouldRetry(ctx, resp, err)
+	})
+	if err != nil {
+		return err
+	}
+	o.modTime = modTime
+	return nil
+}
+
+// Storable says whether this object can be stored.
+func (o *Object) Storable() bool {
+	return true
+}
+
+// Open an object for reading.
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (io.ReadCloser, error) {
+	parameters := api.NewQueryParameters()
+	resolvedPath := o.fs.resolvePath(o.remote)
+	parameters.SetPath(resolvedPath)
+
+	fs.FixRangeOption(options, o.Size())
+	opts := rest.Opts{
+		Method:     "GET",
+		Path:       "/file",
+		Parameters: parameters.Values,
+		Options:    options,
+	}
+	var resp *http.Response
+	var err error
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return o.fs.shouldRetry(ctx, resp, err)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, err
+}
+
+// Update the existing object
+// with the contents of the io.Reader, modTime and size.
+//
+// For unknown-sized contents (indicated by src.Size() == -1)
+// this will try to properly upload it in multiple chunks.
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	remote := src.Remote()
+	modTime := src.ModTime(ctx)
+	fs.Debugf(o.fs, "executing Update(%s, %v).", remote, options)
+	resolvedPath := o.fs.resolvePath(remote)
+
+	if o.fs.tokenRenewer != nil {
+		o.fs.tokenRenewer.Start()
+		defer o.fs.tokenRenewer.Stop()
+	}
+
+	// PutUnchecked can pass a valid SeekOption to skip ahead.
+	var offset uint64
+	for _, option := range options {
+		if seekoption, ok := option.(*fs.SeekOption); ok {
+			offset = uint64(seekoption.Offset)
+			break
+		}
+	}
+
+	var info *api.HiDriveObject
+	var err, metaErr error
+	if offset > 0 || src.Size() == -1 || src.Size() >= int64(o.fs.opt.UploadCutoff) {
+		fs.Infof(o.fs, "Uploading with chunks of size %v and %v transfers in parallel at path '%s'.", int(o.fs.opt.UploadChunkSize), o.fs.opt.UploadConcurrency, resolvedPath)
```

Comment:
This should be a debug - Info logs are for rclone to announce transfers completed.